pkg/controllers/managedzone/managedzone_controller.go

```diff
@@ -38,11 +38,18 @@ const (
 	ManagedZoneFinalizer = "kuadrant.io/managed-zone"
 )
 
+var DNSProvider dns.Provider
+
+type DNS interface {
```

Comment:
I think I would move this to the dns package and provider.go. Also I don't expect we will add anything more to this interface so you could change it just to be ``` type ProviderFactory func(ctx context.Context, managedZone *v1alpha1.ManagedZone) (Provider, error) ```

Reply:
Then in the controllers you can set ```dnsProviderFactory dns.ProviderFactory``` in the struct